src/chk/chk_leader.c

```diff
@@ -1635,6 +1635,91 @@ chk_leader_handle_pools_svc(struct chk_sched_args *csa)
 	return rc;
 }
 
+static int
+chk_leader_pool_mbs_one(struct chk_instance *ins, struct chk_pool_rec *cpr)
+{
+	struct rsvc_client	 client = { 0 };
+	crt_endpoint_t		 ep = { 0 };
+	struct rsvc_hint	 hint = { 0 };
+	struct chk_bookmark	*cbk = &ins->ci_bk;
+	d_rank_list_t		*ps_ranks = NULL;
+	struct chk_pool_shard	*cps;
+	struct ds_pool_clue	*clue;
+	int			 rc = 0;
+	int			 rc1;
+	int			 i = 0;
+
+	D_ASSERT(cpr->cpr_mbs == NULL);
+
+	D_ALLOC_ARRAY(cpr->cpr_mbs, cpr->cpr_shard_nr);
+	if (cpr->cpr_mbs == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
```

Comment:
On this error path we'll finalize an rsvc_client that has not been initialized. Could we avoid doing this even if it might work at the moment, please?

Reply:
OK, I will refresh the patch.